src/serializer/ResidualHeapVisitor.js

```diff
@@ -780,9 +783,86 @@ export class ResidualHeapVisitor {
     this.visitValue(val.$ProxyHandler);
   }
 
+  _visitAbstractValueConditional(val: AbstractValue): void {
+    let condition = val.args[0];
+    invariant(condition instanceof AbstractValue);
+
+    let cf = this.conditionalFeasibility.get(val);
+    if (cf === undefined) this.conditionalFeasibility.set(val, (cf = { t: false, f: false }));
+
+    let feasibleT, feasibleF;
+    let savedPath = this.realm.pathConditions;
+    try {
+      this.realm.pathConditions = this.scope instanceof Generator ? this.scope.pathConditions : [];
+
+      let impliesT = Path.implies(condition);
+      let impliesF = Path.impliesNot(condition);
+      invariant(!(impliesT && impliesF));
+
+      if (!impliesT && !impliesF) {
+        feasibleT = feasibleF = true;
+      } else {
+        feasibleT = impliesT;
+        feasibleF = impliesF;
+      }
+    } finally {
+      this.realm.pathConditions = savedPath;
+    }
+
+    let visitedT = false,
+      visitedF = false;
+
+    if (!cf.t && feasibleT) {
+      val.args[1] = this.visitEquivalentValue(val.args[1]);
+      cf.t = true;
+      if (cf.f) val.args[0] = this.visitEquivalentValue(val.args[0]);
+      visitedT = true;
+    }
+
+    if (!cf.f && feasibleF) {
+      val.args[2] = this.visitEquivalentValue(val.args[2]);
+      cf.f = true;
+      if (cf.t) val.args[0] = this.visitEquivalentValue(val.args[0]);
+      visitedF = true;
+    }
+
+    if (!visitedT || !visitedF) {
+      let fixpoint_rerun = () => {
+        let progress = false;
+        invariant(cf !== undefined);
+        if (cf.f && cf.t) {
+          invariant(!visitedT || !visitedF);
+          this.visitValue(val.args[0]);
+        }
+
+        if (cf.t && !visitedT) {
+          this.visitValue(val.args[1]);
+          progress = visitedT = true;
+        }
+        invariant(cf.t === visitedT);
+
+        if (cf.f && !visitedF) {
+          this.visitValue(val.args[2]);
+          progress = visitedF = true;
+        }
+        invariant(cf.f === visitedF);
+
+        if (!visitedT || !visitedF) this._withUnrelatedScope(this.scope, fixpoint_rerun);
```

Comment:
I have no idea why this works. You should add some comments to describe the basic idea in some detail.

Reply:
Will add comments.